packages/cmf/src/selectors/components.js

```diff
@@ -0,0 +1,41 @@
+import getToJSMemoized from './toJS';
+
+/**
+ * return a collection or subset of a collection from a cmf store
+ * this function returns collection from cmf.components, where the local state of some components
+ * @param {Object} state
+ * @param {String or Array<String>} collectionPath
+ * @example
+ *  get('foo.bar', true) === state.cmf.components.getIn(['foo', 'bar'], true)
+ */
+export function get(state, collectionPath, defaultValue) {
+	let path;
+	if (typeof collectionPath === 'string') {
+		path = collectionPath.split('.');
+	} else if (Array.isArray(collectionPath)) {
+		path = collectionPath;
+	}
+	if (path) {
+		return state.cmf.components.getIn(path, defaultValue);
+	}
+	throw Error(`Type mismatch: collectionPath should be a string or an array of string
+got ${collectionPath}`);
```

Comment:
Should use a switch case without the extra `path` variable and with the default case throwing the error.

Reply:
Hello @talend-glorieux , sorry, seems I didn't quite get what you mean

We use two different expressions in both if cases (typeof and Array.isArray checking), the switch expression is evaluated once, so what is the point to use switch here?